application/controllers/UserManagementController.php

```diff
@@ -1106,35 +1121,38 @@ public function actionBatchAddGroup()
             );
         }
         $aItems = json_decode(Yii::app()->request->getPost('sItems', []));
-        $iUserGroupId = Yii::app()->request->getPost('addtousergroup');
-
-        if ($iUserGroupId) {
-            $oUserGroup = UserGroup::model()->findByPk($iUserGroupId);
-            $aResults = [];
-
-            foreach ($aItems as $sItem) {
-                $aResults[$sItem]['title'] = '';
-                $model = $this->loadModel($sItem);
-                $aResults[$sItem]['title'] = $model->users_name;
-                if (!$oUserGroup->hasUser($sItem)) {
-                    $aResults[$sItem]['result'] = $oUserGroup->addUser($sItem);
-                } else {
-                    $aResults[$sItem]['result'] = false;
-                    $aResults[$sItem]['error'] = gT('User is already a member of the group.');
-                }
-            }
-        } else {
-            foreach ($aItems as $sItem) {
-                $aResults[$sItem]['title'] = '';
-                $model = $this->loadModel($sItem);
-                $aResults[$sItem]['title'] = $model->users_name;
+        $iUserGroupId = intval(App()->request->getPost('addtousergroup'));
+        $oUserGroup = UserGroup::model()->findByPk($iUserGroupId);
+
+        if (!$iUserGroupId || !$oUserGroup) {
+            return $this->renderPartial(
+                'partial/error',
+                ['errors' => [gT("Group not found")], 'noButton' => true]
+            );
+        }
+        /* see UserGroupController->checkBeforeAddDeleteUser */
+        if (!Permission::model()->hasGlobalPermission('superadmin', 'read') && $oUserGroup->owner_id != App()->getCurrentUserId()) {
+            return $this->renderPartial(
```

Comment:
Why not a 403?

Reply:
Same than the existing test before https://github.com/LimeSurvey/LimeSurvey/blob/2d70abf1773cdb910d4cec7fa528f200d37dcd8d/application/controllers/UserManagementController.php#L1117